### ISSUE TITLE
[FEATURE] Switch theme depending on browser configuration without JS

### ISSUE THREAD

**Issue by Ppjet6** [onweru/compose#92]
**I have checked all the prerequisites below and I don't see this is available**

- [x] Read the README.md
- [x] Have the extended version of Hugo installed
- [x] Used the exampleSite's config.toml as a reference

**Describe the feature**

My website has the light theme enabled by default but I use the dark theme. When disabling JS for unrelated testing, I realized I don't have access to the dark theme anymore.

My browser is configured to prefer dark themes and I know this preference is used in other websites.
This would allow degrading (no JS) to still permit theme switching.

**Desktop (please complete the following information):**
 - OS: Linux
 - Browser: Firefox nightly 111.0a1

**Comment by ltGuillaume** [comment]
Here's a bunch of examples: https://discourse.gohugo.io/t/language-switcher-in-menu/11570/

**Comment by onweru** [comment]
@Ppjet6,

I have explored this one. While I have applied simple fix for this, it will not help beyond the theme switching.

If javascript is disabled at the browser level, at least these other functions won't work 👇🏼 

1. Mobile menu
2. Docs sidebar 
3. Color mode switching
4. Search
5. Buttons' spacing.

Of these, only #1 & #5 I could see myself considering supporting where javascript is disabled  in future.
